### DISCUSSION TITLE
Not able to save Question Set of Observation with rubrics in Dock dev

### DISCUSSION THREAD

**Discussion by ramkumar-ns-dev** [Sunbird-coKreat/Community#96] [Q&A]
### **Prerequisites**
I have searched for existing issues that already report this problem, without success.

### **Current Behaviour**
I'm trying to save Question set as Draft of Observation with rubrics in dock dev but it's throwing 400 status as Items not exist.

### **Expected Behaviour**
When user try to save Question set as Draft it should need to save Question set without failure.

### **Steps to Reproduce**
	1. Created project from Sourcing admin.
	2. Submitted nomination from contributor.
	3. Approve Nomination from sourcing admin.
	4. Click "Create/upload content" on project page it will open Question set page with domain and criteria.
	5. When we click on "save as Draft" button instead of saving the Question set it's failing from following API.
	6. Failing API => https://dock.sunbirded.org/api/question/v1/list
	
### **API Endpoints used**
       1. https://dock.sunbirded.org/api/question/v1/list
        
### Additional Information
Tested on Google Chrome: Version 114.0.5735.198 (Official Build) (arm64)

**Note** : We have created new category and terms called "Topic" and "Science" in TPD framework afterwards this issue happening.

can anybody help on this.
@vaivk369 

cc: @kiranharidas187

**Comment by krgauraw** [comment]
@ramkumar-pacewisdom Can you please provide exact error details of question list api.
cc: @kiranharidas187 @pooja

**Reply by ramkumar-ns-dev** [reply]
@krgauraw added the exact details, kindly let me know if more details needed.
**Request :** 
     https://dock.sunbirded.org/api/question/v1/list
**Payload :** 
    {
      "request": {
          "search": {
            "identifier": [
                "tpd_topic_science",
                "tpd_topic_social-science"
               ]
            }
       }
   }
**Response :** 
    {"id":"api.questions.list","ver":"3.0","ts":"2023-10-13T11:20:22ZZ","params":{"resmsgid":"78c7b87a-ba2c-4474-b3a5-bca725241b1d","msgid":null,"err":"ERR_BAD_REQUEST","status":"failed","errmsg":"Request contains invalid identifiers : [tpd_topic_science, tpd_topic_social-science]."},"responseCode":"CLIENT_ERROR","result":{"messages":null}}

**Reply by krgauraw** [reply]
Hi @ramkumar-pacewisdom the request payload is having wrong data. Ideally the object identifiers (do_id) should be passed as an input for list api. So its not an api issue. Can you please check, why wrong data is getting passed to this api?

@vaivk369 Can you help here.

cc: @kiranharidas187 @pooja

**Reply by ramkumar-ns-dev** [reply]
@krgauraw the ID is coming from the API is ("tpd_topic_science") like this only, not as ("do_id"). This ID's are coming from "
https://dock.sunbirded.org/api/framework/v1/read/TPD" API and i've added the following fragment of the API response.

`{
    "identifier": "tpd_topic_science",
    "code": "science",
    "translations": null,
    "name": "Science",
    "description": "Science",
    "index": 1,
    "category": "topic",
    "status": "Live"
}`

@vaivk369 please let me know if i'm wrong.
